client/resoucemanager_client.go

```diff
@@ -0,0 +1,265 @@
+	if cc, ok := c.clientConns.Load(c.GetLeaderAddr()); ok {
+		return rmpb.NewResourceManagerClient(cc.(*grpc.ClientConn))
+	}
+	return nil
+}
+
+// ListResourceGroups loads and returns target keyspace's metadata.
+func (c *client) ListResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, error) {
+	req := &rmpb.ListResourceGroupsRequest{}
+	resp, err := c.resouceManagerClient().ListResourceGroups(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	resErr := resp.GetError()
+	if resErr != nil {
+		return nil, errors.Errorf("[pd]" + resErr.Message)
+	}
+	return resp.GetGroups(), nil
+}
+
+func (c *client) GetResourceGroup(ctx context.Context, resourceGroupName string) (*rmpb.ResourceGroup, error) {
+	req := &rmpb.GetResourceGroupRequest{
+		ResourceGroupName: resourceGroupName,
+	}
+	resp, err := c.resouceManagerClient().GetResourceGroup(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	resErr := resp.GetError()
+	if resErr != nil {
+		return nil, errors.Errorf("[pd]" + resErr.Message)
+	}
+	return resp.GetGroup(), nil
+}
+
+func (c *client) AddResourceGroup(ctx context.Context, resourceGroupName string, settings *rmpb.GroupSettings) (string, error) {
+	return c.putResourceGroup(ctx, resourceGroupName, settings, 0 /* type add resource group */)
+}
+
+func (c *client) ModifyResourceGroup(ctx context.Context, resourceGroupName string, settings *rmpb.GroupSettings) (string, error) {
+	return c.putResourceGroup(ctx, resourceGroupName, settings, 1 /* type modify resource group */)
+}
+
+func (c *client) putResourceGroup(ctx context.Context, resourceGroupName string, settings *rmpb.GroupSettings, typ int) (str string, err error) {
+	group := &rmpb.ResourceGroup{
+		Name:     resourceGroupName,
+		Settings: settings,
+	}
+	req := &rmpb.PutResourceGroupRequest{
+		Group: group,
+	}
+	var resp *rmpb.PutResourceGroupResponse
+	if typ == 0 {
+		resp, err = c.resouceManagerClient().AddResourceGroup(ctx, req)
+	} else {
+		resp, err = c.resouceManagerClient().ModifyResourceGroup(ctx, req)
+	}
+	if err != nil {
+		return str, err
+	}
+	resErr := resp.GetError()
+	if resErr != nil {
+		return str, errors.Errorf("[pd]" + resErr.Message)
+	}
+	str = resp.GetBody()
+	return
+}
+
+func (c *client) DeleteResourceGroup(ctx context.Context, resourceGroupName string) (string, error) {
+	req := &rmpb.DeleteResourceGroupRequest{
+		ResourceGroupName: resourceGroupName,
+	}
+	resp, err := c.resouceManagerClient().DeleteResourceGroup(ctx, req)
+	if err != nil {
+		return "", err
+	}
+	resErr := resp.GetError()
+	if resErr != nil {
+		return "", errors.Errorf("[pd]" + resErr.Message)
+	}
+	return resp.GetBody(), nil
+}
+
+func (c *client) AcquireTokenBuckets(ctx context.Context, request *rmpb.TokenBucketsRequest) ([]*rmpb.TokenBucketResponse, error) {
+	req := &tokenRequest{
+		done:       make(chan error, 1),
+		requestCtx: ctx,
+		clientCtx:  c.ctx,
+	}
+	req.Requeset = request
+	c.tokenDispatcher.tokenBatchController.tokenRequestCh <- req
```

Comment:
It seems that this will easily be blocked since the `tokenRequestCh`'s capacity is 1.

Reply:
Currently we have not implemented batch, so it will also be blocked by `Wait()`？